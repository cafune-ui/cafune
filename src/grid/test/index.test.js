import Grid from '../';
const { Item } = Grid;
import { render } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

const IconList = [
  'img',
  'notice',
  'arrow-right',
  'arrow-left',
  'check',
  'wrong',
  'more'
];
describe('Grid', () => {
  it('should render with 5 column', () => {
    const wrapper = deep(
      <Grid column={5}>
        {IconList.map(item => (
          <Item key={item}>
            <span>{item}</span>
          </Item>
        ))}
      </Grid>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with no border', () => {
    const wrapper = deep(
      <Grid border={false}>
        {IconList.map(item => (
          <Item key={item}>
            <span>{item}</span>
          </Item>
        ))}
      </Grid>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with no center', () => {
    const wrapper = deep(
      <Grid center={false} gutter={20}>
        {IconList.map(item => (
          <Item key={item}>
            <span>{item}</span>
          </Item>
        ))}
      </Grid>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with gutter', () => {
    const wrapper = deep(
      <Grid square={false} gutter={20}>
        {IconList.map(item => (
          <Item key={item}>
            <span>{item}</span>
          </Item>
        ))}
        <div>test</div>
      </Grid>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.render(
      <Grid square={false}>
        {IconList.map(item => (
          <Item key={item}>
            <span>{item}</span>
          </Item>
        ))}
      </Grid>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render nothing', () => {
    const wrapper = shallow(<Grid />);
    expect(wrapper).toMatchSnapshot();
  });
});
