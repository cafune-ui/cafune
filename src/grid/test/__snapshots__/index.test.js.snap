// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render nothing 1`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Grid should render with 5 column 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "column": 5,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "ind": 1,
          "key": "img",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "img",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "img",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 2,
          "key": "notice",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "notice",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "notice",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 3,
          "key": "arrow-right",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-right",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-right",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 4,
          "key": "arrow-left",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-left",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-left",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 5,
          "key": "check",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "check",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "check",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 6,
          "key": "wrong",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "wrong",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "wrong",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 7,
          "key": "more",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "more",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "more",
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Grid should render with gutter 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "gutter": 20,
      "square": false,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "ind": 1,
          "key": "img",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "img",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "img",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 2,
          "key": "notice",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "notice",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "notice",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 3,
          "key": "arrow-right",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-right",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-right",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 4,
          "key": "arrow-left",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-left",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-left",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 5,
          "key": "check",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "check",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "check",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 6,
          "key": "wrong",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "wrong",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "wrong",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 7,
          "key": "more",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "more",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "more",
        "nodeName": [Function],
      },
      VNode {
        "attributes": undefined,
        "children": Array [
          "test",
        ],
        "key": undefined,
        "nodeName": "div",
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Grid should render with gutter 2`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "square": false,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "ind": 1,
          "key": "img",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "img",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "img",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 2,
          "key": "notice",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "notice",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "notice",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 3,
          "key": "arrow-right",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-right",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-right",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 4,
          "key": "arrow-left",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-left",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-left",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 5,
          "key": "check",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "check",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "check",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 6,
          "key": "wrong",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "wrong",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "wrong",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 7,
          "key": "more",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "more",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "more",
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Grid should render with no border 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "border": false,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "ind": 1,
          "key": "img",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "img",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "img",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 2,
          "key": "notice",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "notice",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "notice",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 3,
          "key": "arrow-right",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-right",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-right",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 4,
          "key": "arrow-left",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-left",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-left",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 5,
          "key": "check",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "check",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "check",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 6,
          "key": "wrong",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "wrong",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "wrong",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 7,
          "key": "more",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "more",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "more",
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Grid should render with no center 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "center": false,
      "gutter": 20,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "ind": 1,
          "key": "img",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "img",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "img",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 2,
          "key": "notice",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "notice",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "notice",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 3,
          "key": "arrow-right",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-right",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-right",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 4,
          "key": "arrow-left",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "arrow-left",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "arrow-left",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 5,
          "key": "check",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "check",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "check",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 6,
          "key": "wrong",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "wrong",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "wrong",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "ind": 7,
          "key": "more",
        },
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              "more",
            ],
            "key": undefined,
            "nodeName": "span",
          },
        ],
        "key": "more",
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
